mlir/lib/Quantum/Transforms/annotate_function.cpp

```diff
@@ -0,0 +1,189 @@
+// Copyright 2023 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "Quantum/IR/QuantumOps.h"
+#include "mlir/Analysis/CallGraph.h"
+#include "mlir/Conversion/LLVMCommon/TypeConverter.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+#include "Catalyst/IR/CatalystOps.h"
+#include "Quantum/Transforms/Passes.h"
+#include "Quantum/Transforms/Patterns.h"
+
+using namespace mlir;
+using namespace catalyst::quantum;
+
+namespace {
+
+static constexpr const char *hasMeasureAttrName = "catalyst.invalidGradientOperation";
```

Comment:
1. If we can't encode it in the *td files, let me suggest to move this string constant to some header to only declare it once.
2. Wouldn't it better to name this constant `invalidGradientOperationAttrName` ?

Reply:
https://github.com/PennyLaneAI/catalyst/pull/204/commits/c1a8915f44ee3fc1b6e21ba8a40b8b44298ee3ba https://github.com/PennyLaneAI/catalyst/pull/204/commits/4756e695fa567b96516dc0488f714878e51d83a3